src/core/dom/layout/viewport-observer.js

```diff
@@ -31,7 +29,7 @@ export function createViewportObserver(ioCallback, win, opts = {}) {
 /** @type {!WeakMap<!Window, !IntersectionObserver>} */
 const viewportObservers = new WeakMap();
 
-/** @type {!WeakMap<!Element, function(boolean)>} */
+/** @type {!WeakMap<!Element, !Array<function(IntersectionObserverEntry)>>} */
```

Comment:
Do we need to update all the current usages of `observeWithSharedInOb` to handle an InObEntry vs. true/false?

Reply:
This is a good point.

Reply:
Boo for lack of type checking 👎  (in non-core)

Reply:
😢